taipy/core/_orchestrator/_orchestrator.py

```diff
@@ -134,7 +145,11 @@ def submit_task(
         else:
             if wait:
                 cls._wait_until_job_finished(job, timeout=timeout)
-        return job
+        return submission
+
+    @classmethod
+    def _update_submission_status(cls, job: Job):
+        cls._submission_entities[job.submit_id]._update_submission_status(job)
```

Comment:
I decided to store `Submission`s in `cls._submission_entities` instead of getting it from the repo is because `Submission` has some in-memory state that are not stored in the repo (`__blocked_jobs`, `__pending_jobs`, `__running_jobs`). We can make changes to store them in the repo and replace `cls._submission_entities` with `SubmissionManagerFactory._build_manager()._get(job.submit_id)` but I think that's outside the scope of this ticket so I decided not to go forward with it.

Reply:
Agree, I think it's about time as well since we're implementing cluster mode. Can't delay it any further. Though I can't remember all the places that have this issues :/

Reply:
@jrobinAV 
I have created the ticket here. Let me know what you think? https://github.com/Avaiga/taipy/issues/660